### PR TITLE
Implement retaliation

### PR DESCRIPTION
**Pull request by TeoTwawki**
https://github.com/DarkstarProject/darkstar/pull/943/files?w=1

### CODE REVIEW

**Comment by maxtherabbit** [comment]
looks right

File: scripts/globals/abilities/retaliation.lua [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+function OnAbilityCheck(player,target,ability)
+    return 0,0;
+end;
+
+-----------------------------------
+-- OnUseAbility
+-----------------------------------
+
+function OnUseAbility(player, target, ability)
+    player:delStatusEffectSilent(EFFECT_RETALIATION);
```
Comment (teschnei) [inline scripts/globals/abilities/retaliation.lua]: not needed (that's what all the flags in status_effects.sql are for)

File: src/map/utils/battleutils.cpp [hunk 2]
```diff
@@ -551,7 +551,56 @@ bool HandleSpikesDamage(CBattleEntity* PAttacker, CBattleEntity* PDefender, apAc
...
+                dmg = dmg + bonus;
+
+                // FINISH HIM! dun dun dun
+                // TP and stoneskin are handled inside TakePhysicalDamage
+                Action->spikesMessage = 536;
+                Action->spikesParam = battleutils::TakePhysicalDamage(PDefender, PAttacker, dmg, false, SLOT_MAIN, 1, NULL, true);
+                PAttacker->addHP(-Action->spikesParam);
+            }
+            /*
+            else
```
Comment (teschnei) [inline src/map/utils/battleutils.cpp]: just put the skill up code in the if TYPE_PC block? pretty sure the rest of that is all fine for mobs, then you don't have to worry about it being not implemented later

**Comment by teschnei** [comment]
a good idea would be to write a new function that calculates the damage of one attack and have all this just call that (get weapon dmg, fstr, ratio, check for null damage, check for absorbed damage, etc)

**Comment by TeoTwawki** [comment]
Updated. Couldn't do the new function, need to much needed that is unrelated to this JA. Would like to revisit that idea later though.

**Pull request merged**